Sources/Kitura/CodableRouter.swift

```diff
@@ -600,21 +603,38 @@ extension Router {
         }
     }
 
-    internal func parameterIsPresent(in route: String) -> Bool {
-        if route.contains(":") {
-            let paramaterString = route.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
-            let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
-            Log.error("Erroneous path '\(route)', parameter ':\(parameter)' is not allowed. Codable routes do not allow parameters.")
+    /// Precondition: Path is known to contain a : character
+    func pathHasSingleParamIdAsSuffix(_ path: String) -> Bool {
+        let paramaterString = path.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
+        let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
+        return parameter.isEmpty || parameter.elementsEqual("id")
+    }
+
+    func pathSyntaxIsValid(_ path: String, identifierExpected: Bool) -> Bool {
+        let identifierSupplied = path.contains(":")
```

Comment:
@djones6 I have re-ordered the checks based on the ```identifierExpected``` flag as  it is vain to do the split operation if the identifier itself is not expected for the route